solidity.pegjs

```diff
@@ -1424,8 +1456,18 @@ SourceElements
     }
 
 SourceElement
-  = AssignmentExpression __ EOS
-  / DeclarativeExpression __ EOS
+  = exp:AssignmentExpression  __ EOS &{
+      return (
+        exp.type === 'AssignmentExpression' && 
+        exp.left.type === 'DeclarativeExpression' && 
+        exp.left.storage_location === null
+      )
+    }{

```

Comment:
I agree with you. It's ugly. But look down the AssignmentExpression path. Won't enforcing this distinction involve something like the above or writing separate path that reaches quite a way down?